ci-operator/step-registry/aws/provision/public-to-private/aws-provision-public-to-private-commands.sh

```diff
@@ -0,0 +1,162 @@
+ocp_major_version=$( echo "${ocp_version}" | awk --field-separator=. '{print $1}' )
+ocp_minor_version=$( echo "${ocp_version}" | awk --field-separator=. '{print $2}' )
+
+if (( ocp_minor_version <= 11&& ocp_major_version == 4 )); then
+  echo "CPMS support for AWS was added in 4.12, the following step is not applicable for this OCP version, quit now."
+  exit 1
+fi
+
+if [ -f "${SHARED_DIR}/kubeconfig" ] ; then
+    export KUBECONFIG=${SHARED_DIR}/kubeconfig
+else
+    echo "ERROR: fail to get the kubeconfig file under ${SHARED_DIR}!!"
+    exit 1
+fi
+
+function run_command() {
+    local CMD="$1"
+    echo "Running command: ${CMD}"
+    eval "${CMD}"
+    echo ""
+}
+
+function check_cpms_status() {
+    DesiredCount=$(oc get controlplanemachineset/cluster -n openshift-machine-api -o=jsonpath='{.spec.replicas}')
+    COUNTER=0
+    #The rolling update and re-creation of the three masters will take about 1 hour
+    while [ $COUNTER -lt 4500 ]
+    do
+        sleep 300
+        COUNTER=`expr $COUNTER + 300`
+        echo "waiting ${COUNTER}s"
+	#During master rolling update, the API server may not response
+	readyCPMSCount=$(oc get controlplanemachineset/cluster -n openshift-machine-api -o=jsonpath='{.status.readyReplicas}' || true)
+	updatedCPMSCount=$(oc get controlplanemachineset/cluster -n openshift-machine-api -o=jsonpath='{.status.updatedReplicas}' || true)
+        if [[ ${readyCPMSCount} == "${DesiredCount}" ]] && [[ ${updatedCPMSCount} == "${DesiredCount}" ]]; then
+            echo "CPMS update finished."
+            break
+        fi
+    done
+    if [[ ${readyCPMSCount} != "${DesiredCount}" ]] || [[ ${updatedCPMSCount} != "${DesiredCount}" ]]; then
+	echo "Something wrong in the CPMS update:"
+        run_command "oc get machines -n openshift-machine-api"
+        run_command "oc get controlplanemachineset/cluster -n openshift-machine-api -o yaml"
+        return 1
+    fi
+}
+
+
+#Setting the Ingress Controller to private
+echo "Configuring DNS records to be published in the private zone"
+run_command "oc patch dnses.config.openshift.io/cluster --type=merge --patch='{\"spec\": {\"publicZone\": null}}'"
+
+echo "Setting the Ingress Controller to private"
+oc replace --force --wait --filename - <<EOF
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  namespace: openshift-ingress-operator
+  name: default
+spec:
+  endpointPublishingStrategy:
+    type: LoadBalancerService
+    loadBalancer:
+      scope: Internal
+EOF
+
+
+# Remove the external load balancers from CPMS
+CPMS_PATCH=$(mktemp)
+cat >"${CPMS_PATCH}" <<EOF
+spec:
+  template:
+    machines_v1beta1_machine_openshift_io:
+      spec:
+        providerSpec:
+          value:
+            loadBalancers:
+            - name: ${INFRA_ID}-int
+              type: network
+EOF
+
+echo "Remove the external load balancers from CPMS"
+run_command "oc patch controlplanemachineset/cluster -n openshift-machine-api --type merge --patch-file ${CPMS_PATCH}"
```

Comment:
This merge will remove `${INFRA_ID}-ext` setting, right?

Reply:
yes, with only ${INFRA_ID}-int LB configured in the CPMS